File: itunes_search_api.rb
#!/usr/bin/env ruby

require 'net/http'

uri = URI('https://itunes.apple.com/lookup')
params = { :id=> 909253 }
uri.query = URI.encode_www_form(params)

Net::HTTP.start(uri.host, uri.port,
  :use_ssl => uri.scheme == 'https') do |http|
  request = Net::HTTP::Get.new uri

  response = http.request request # Net::HTTPResponse object
  puts response.body
end

File: itunes_search_api.pl
#!/usr/bin/perl

use LWP::UserAgent;
use URI;
use utf8;

my $id = '909253';
my $uri = URI->new('https://itunes.apple.com/lookup');
my $params = {
    id => $id
};

$uri->query_form(%$params);
my $ua = LWP::UserAgent->new;
#my $res = $ua->get($uri);
my $res = $ua->post($uri,\@params);

print $uri;
if($res->is_success){
  print $res->content;
}else{
  die $res->status_line;
}
